govcd/generic_functions.go

```diff
@@ -38,3 +40,206 @@ func oneOrError[T any](key, name string, entitySlice []*T) (*T, error) {
 
 	return entitySlice[0], nil
 }
+
+// genericCreateBareEntity implements a common pattern for creating an entity throughout codebase
```

Comment:
I have moved them to `openapi_generic.go`. Didn't move to `openapi.go` mainly because they contain public functions so it is easy to give someone a reference.